src/prefect/server/api/flow_runs.py

```diff
@@ -520,27 +520,142 @@ async def set_flow_run_state(
+
+    try:
+        parsed_run_input = orjson.loads(run_input)
+    except orjson.JSONDecodeError as exc:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=f"Run input was not valid JSON: {exc.msg}",
+        )
+
+    try:
+        jsonschema.validate(parsed_run_input, schema)
+    except (jsonschema.ValidationError, jsonschema.SchemaError) as exc:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=f"Run input did not match expected schema: {exc.message}",
+        )
+
+    if state.state_details.pause_reschedule:
+        orchestration_result = await models.flow_runs.set_flow_run_state(
+            session=session,
+            flow_run_id=flow_run.id,
+            state=schemas.states.Scheduled(
+                name="Resuming", scheduled_time=pendulum.now("UTC")
+            ),
+            flow_policy=flow_policy,
+            orchestration_parameters=orchestration_parameters,
+        )
+    else:
+        orchestration_result = await models.flow_runs.set_flow_run_state(
+            session=session,
+            flow_run_id=flow_run.id,
+            state=schemas.states.Running(),
+            flow_policy=flow_policy,
+            orchestration_parameters=orchestration_parameters,
+        )
+
+    if orchestration_result.status != schemas.responses.SetStateStatus.ACCEPT:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=(
+                f"Could not resume {state.type} flow run:"
+                f" {orchestration_result.details.reason}"
+            ),
+        )
+
+    # If the resume state change was accepted, store the flow run input.
+    return schemas.core.FlowRunInput(
+        flow_run_id=flow_run.id,
+        key=keyset["response"],
+        value=orjson.dumps(parsed_run_input).decode("utf-8"),
+    )
+
+
 @router.post("/{id}/input", status_code=status.HTTP_201_CREATED)
 async def create_flow_run_input(
     flow_run_id: UUID = Path(..., description="The flow run id", alias="id"),
     key: str = Body(..., description="The input key"),
     value: bytes = Body(..., description="The value of the input"),
     sender: Optional[str] = Body(None, description="The sender of the input"),
     db: PrefectDBInterface = Depends(provide_database_interface),
+    flow_policy: BaseOrchestrationPolicy = Depends(
+        orchestration_dependencies.provide_flow_policy
+    ),
+    orchestration_parameters: dict = Depends(
+        orchestration_dependencies.provide_flow_orchestration_parameters
+    ),
 ):
     """
     Create a key/value input for a flow run.
+
+    If the flow run is paused or suspended, the input will be validated against
+    the run input schema that the flow run expects. If the input matches and
+    validates, the flow run will resume. If it matches and doesn't validate,
+    the client will receive an error.
+
+    If the flow run was not paused or suspended, or if the input is not what
```

Comment:
>   If the flow run was not paused or suspended...

Maybe clearer to say something like "Flow runs that aren't paused or suspended are stored without validation. If the input doesn't match the expected schema, it is stored but the flow run will not resume."

So we just store it in the db, but don't do anything with it, right?